src/plugins/intel_gpu/src/graph/impls/implementation_map.hpp

```diff
@@ -162,23 +146,20 @@ class implementation_map {
                 return factory;
             }
         }
-        std::stringstream target_impl_type_ss;
-        target_impl_type_ss << preferred_impl_type;
-        throw std::runtime_error(std::string("implementation_map for ") + typeid(primitive_kind).name() +
-                                    " could not find any implementation to match key: " +
-                                    get_key_name(key) + ", impl_type: " + target_impl_type_ss.str() + ", node_id: " + impl_params.desc->id);
+        OPENVINO_ASSERT(false, "[GPU] implementation_map for ", typeid(primitive_kind).name(),
+                               " could not find any implementation to match key: ", get_key_name(key),
+                               ", impl_type: ", preferred_impl_type, ", node_id: ",  impl_params.desc->id);
     }
 
     // check if for a given engine and type there exist an implementation
-    static bool check(const typed_program_node<primitive_kind>& primitive, const kernel_impl_params& impl_params) {
-        impl_types target_impl_type = primitive.get_preferred_impl_type();
-        auto key = key_builder()(impl_params.input_layouts[0]);
+    static bool check(const kernel_impl_params& impl_params, impl_types target_impl_type) {
+        auto input_layout = !impl_params.input_layouts.empty() ? impl_params.input_layouts[0] : layout{ov::PartialShape{}, data_types::f32, format::any};
+        auto key = key_builder()(input_layout);
```

Comment:
What kind of the case that the input_layout can be empty? 
Shouldn't it be an assert?

Reply:
`input_layout` primitive, for instance. For such primitives we have special key_builder which actually returns `-1`, so we don't need to throw exception here.